### PR TITLE
Pigeon 21.2 upgrade

### PR DESCRIPTION
**Pull request by diegotori**
Upgrades Pigeon to version 21.2. Also upgrades minimum Flutter/Dart versions to 3.22.0/3.4.0 respectively.

Prep work for eventual `1.3.x` release.

### CODE REVIEW

File: wakelock_plus/pubspec.yaml [hunk 1]
```diff
@@ -19,7 +19,7 @@ dependencies:
 
   # Windows dependencies
   # win32 is compatible across v4 and v5 for Win32 only (not COM)
```
Comment (vbuberen) [inline wakelock_plus/pubspec.yaml:21]: nit: I think this comment can be dropped, considering that v4 isn't allowed anymore.

File: wakelock_plus_platform_interface/pubspec.yaml [hunk 2]
```diff
@@ -7,8 +7,8 @@ repository: >-2
   https://github.com/fluttercommunity/wakelock_plus/tree/main/wakelock_plus_platform_interface
 
 environment:
-  sdk: '>=3.3.0 <4.0.0'
-  flutter: ">=3.19.0"
+  sdk: '>=3.4.0 <4.0.0'
+  flutter: ">=3.22.0"
```
Comment (vbuberen) [inline wakelock_plus_platform_interface/pubspec.yaml:11]: Not sure I follow the reason of this change - is there a need to do this bump? I see that Pigeon 21.2 is Ok with Dart 3.3 and Flutter 3.19.

**Review by vbuberen — APPROVED** [review]
Looks good to me.

As I have some Kotlin background, looked at it in a more detailed manner and seems like in newer versions there was a set of code style linter suggestions applied :)

File: wakelock_plus_platform_interface/pubspec.yaml (hunk 2 above)
Comment (diegotori) [reply wakelock_plus_platform_interface/pubspec.yaml:11]: `win32` version `5.5.1` requires Dart `3.4.0` in the `wakelock_plus` package, which came out in `3.22`.

Maybe the platform interface doesn't need it, but the main package does.

Comment (diegotori) [reply wakelock_plus_platform_interface/pubspec.yaml]: Anyways, rolled it back for the platform interface.

File: wakelock_plus/pubspec.yaml (hunk 1 above)
Comment (diegotori) [reply wakelock_plus/pubspec.yaml]: Fixed.

File: wakelock_plus_platform_interface/pubspec.yaml (hunk 2 above)
Comment (vbuberen) [reply wakelock_plus_platform_interface/pubspec.yaml]: Understood. Thanks for giving a hint

**Pull request merged**